pkg/cmd/grafana-server/main.go

```diff
@@ -56,16 +58,56 @@ func main() {
 		os.Exit(0)
 	}
 
-	if *profile {
+	ultimateProfileEnabled := *profile
+	ultimateProfilePort := *profilePort
+	profileEnv, exists := os.LookupEnv("GF_PROCESS_PROFILE")
+	if exists {
+		enabled, parseErr := strconv.ParseBool(profileEnv)
+		if parseErr != nil {
+			panic(parseErr)
+		} else {
+			ultimateProfileEnabled = enabled
+			profilePortEnv, exists := os.LookupEnv("GF_PROCESS_PROFILE_PORT")
+			if exists {
+				port, parseErr := strconv.ParseInt(profilePortEnv, 0, 64)
```

Comment:
The most appropriate type for port is [uint16](https://stackoverflow.com/questions/113224/what-is-the-largest-tcp-ip-network-port-number-allowable-for-ipv4#answer-113228) but the [`flag`](https://golang.org/pkg/flag/#FlagSet.Uint) package does not define a uint16 `Flag` so I use `ParseInt` with `bitsize` set to 64 for avoiding further type conversions and errors.